substratest/assets.py

```diff
@@ -420,6 +410,7 @@ def __post_init__(self):
             self.testtuple_keys = []
 
     def list_traintuple(self):
+        self.__post_init__()
```

Comment:
`__post_init__` is part of how dataclasses work. If we're not using dataclasses we shouldn't probably use this name and I really don't think we should call it manually, especially in such a call which isn't supposed to alter the object.